### PR TITLE
feat: added CDC event stream

### PR DESCRIPTION
**Pull request by mattayres**

### CODE REVIEW

File: driver/http.go [hunk 1]
```diff
@@ -460,6 +465,50 @@ func (g *httpStreamReader) close() error {
...
+
+func (g *httpEventStreamReader) read() (Event, error) {
+	var res struct {
+		Result apiHTTP.StreamResponse
+	}
+	if err := g.stream.Decode(&res); err != nil {
+		return nil, HTTPError(err, nil)
+	}
+
+	event, err := jsoniter.Marshal(res.Result.Event)
```
Comment (himank) [inline driver/http.go:500]: do we need to do this?

File: driver/grpc.go [hunk 2]
```diff
@@ -355,3 +356,38 @@ func (g *grpcStreamReader) read() (Document, error) {
...
+	stream api.Tigris_StreamClient
+}
+
+func (g *grpcEventStreamReader) read() (Event, error) {
+	resp, err := g.stream.Recv()
+	if err != nil {
+		return nil, GRPCError(err)
+	}
+
+	event, err := jsoniter.Marshal(resp.Event)
```
Comment (himank) [inline driver/grpc.go:383]: do we need to marshal it?

File: driver/http.go (hunk 1 above)
Comment (mattayres) [reply driver/http.go]: nope, fixed

File: driver/grpc.go (hunk 2 above)
Comment (mattayres) [reply driver/grpc.go]: nope, fixed

**Review by himank — APPROVED** [review]

**Pull request merged**